File: ops/frontend/src/types/trade.ts
/** Trade execution domain model. */

import type { Market } from './market'

export type TradeDirection = 'long' | 'short'

export type TradeStatus = 'open' | 'closed' | 'cancelled'

export interface Trade {
  id: string
  /** Market this trade belongs to. India and International never mix. */
  market: Market
  /** ISO timestamp of entry. */
  time: string
  strategy: string
  machine: string
  broker: string
  account: string
  symbol: string
  direction: TradeDirection
  entry: number
  /** `null` while the position is still open. */
  exit: number | null
  quantity: number
  /** Realised (closed) or unrealised (open) pnl in account currency. */
  pnl: number
  /** Execution latency in milliseconds. */
  latencyMs: number
  /** Holding period in seconds. */
  durationSec: number
  status: TradeStatus
}

File: ops/backend/app/schemas/trade.py
"""Trade schemas."""

from typing import Literal

from pydantic import BaseModel

TradeDirection = Literal["long", "short"]
TradeStatus = Literal["open", "closed", "cancelled"]


class Trade(BaseModel):
    """A single executed trade."""

    id: str
    time: str
    strategy: str
    machine: str
    broker: str
    account: str
    symbol: str
    direction: TradeDirection
    entry: float | None = None
    exit: float | None
    quantity: float | None = None
    pnl: float | None = None
    latencyMs: float | None = None
    durationSec: int | None = None
    status: TradeStatus
